source/agora/cli/gen/main.d

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+
+    Entry point for the agora-gen
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.cli.gen.main;
+
+import scpd.scp.SCPDriver;
+
+import std.stdio;
+import std.traits;
+
+/// Workaround for issue likely related to dub #225,
+/// expects a main() function and invokes it after unittesting.
+version (unittest) void main () { } else:
+
+/*******************************************************************************
+
+    Application entry point
+
+*******************************************************************************/
+
+private int main (string[] args)
+{
+    string[] methods;
+
+    foreach (member; __traits(allMembers, scpd.scp.SCPDriver.SCPDriver))
+    {
+        if ((member == "__dtor") || (member == "__xdtor") || (member == "ValidationLevel")) continue;
+        methods ~= member;
```

Comment:
This code is too specialized for SCPDriver, and `methods` may include fields here and not just methods, because there is no check that the member is a method.